api/src/org/apache/cloudstack/api/command/admin/storage/AddImageStoreS3CMD.java

```diff
@@ -201,15 +209,15 @@ public int hashCode() {
         result = 31 * result + (secretKey != null ? secretKey.hashCode() : 0);
         result = 31 * result + (endPoint != null ? endPoint.hashCode() : 0);
         result = 31 * result + (bucketName != null ? bucketName.hashCode() : 0);
-        result = 31 * result + (httpsFlag != null && httpsFlag == true ? 1 : 0);
+        result = 31 * result + (signer != null ? signer.hashCode() : 0);
```

Comment:
The same here: would be nice to move hashCode() and equals() methods to the end of the file.